Src/ILGPU/Backends/PTX/PTXInstructions.Data.cs

```diff
@@ -389,7 +426,12 @@ partial class PTXInstructions
                 { (UnaryArithmeticKind.SinF, ArithmeticBasicValueType.Float32), "sin.approx.f32" },
                 { (UnaryArithmeticKind.CosF, ArithmeticBasicValueType.Float32), "cos.approx.f32" },
 
+                { (UnaryArithmeticKind.TanhF, ArithmeticBasicValueType.Float16), "tanh.approx.f16" },
+                { (UnaryArithmeticKind.TanhF, ArithmeticBasicValueType.Float32), "tanh.approx.f32" },
```

Comment:
We should register the associated intrinsics within the `Algorithms` library to ensure a correct mapping. However, we should think about additional error messages in case of unsupported intrinsics that are registered but not available for a specific target platform.

Reply:
I was thinking to register a `PTXIntrinsic` with `IntrinsicImplementationMode.GenerateCode`, that would throw a `NotSupportedIntrinsicException`. This would mimic the behaviour with existing intrinsics.

Then, in ILGPU.Algorithms, we would need to modify the `XMath` registration to limit `Tanh` to `sm_75` or higher.